specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2023-01-01-preview/sqlvm.json

```diff
@@ -0,0 +1,2676 @@
+      "type": "object",
+      "properties": {
+        "sqlConnectivityUpdateSettings": {
+          "$ref": "#/definitions/SqlConnectivityUpdateSettings",
+          "description": "SQL connectivity type settings."
+        },
+        "sqlWorkloadTypeUpdateSettings": {
+          "$ref": "#/definitions/SqlWorkloadTypeUpdateSettings",
+          "description": "SQL workload type settings."
+        },
+        "sqlStorageUpdateSettings": {
+          "$ref": "#/definitions/SqlStorageUpdateSettings",
+          "description": "SQL storage update settings."
+        },
+        "additionalFeaturesServerConfigurations": {
+          "$ref": "#/definitions/AdditionalFeaturesServerConfigurations",
+          "description": "Additional SQL feature settings."
+        },
+        "sqlInstanceSettings": {
+          "$ref": "#/definitions/SQLInstanceSettings",
+          "description": "SQL Instance settings."
+        },
+        "azureAdAuthenticationSettings": {
+          "$ref": "#/definitions/AADAuthenticationSettings",
+          "description": "Azure AD authentication Settings."
+        }
+      }
+    },
+    "StorageConfigurationSettings": {
+      "description": "Storage Configurations for SQL Data, Log and TempDb.",
+      "type": "object",
+      "properties": {
+        "sqlDataSettings": {
+          "$ref": "#/definitions/SQLStorageSettings",
+          "description": "SQL Server Data Storage Settings."
+        },
+        "sqlLogSettings": {
+          "$ref": "#/definitions/SQLStorageSettings",
+          "description": "SQL Server Log Storage Settings."
+        },
+        "sqlTempDbSettings": {
+          "$ref": "#/definitions/SQLTempDbSettings",
+          "description": "SQL Server TempDb Storage Settings."
+        },
+        "sqlSystemDbOnDataDisk": {
+          "description": "SQL Server SystemDb Storage on DataPool if true.",
+          "type": "boolean"
+        },
+        "diskConfigurationType": {
+          "description": "Disk configuration to apply to SQL Server.",
+          "enum": [
+            "NEW",
+            "EXTEND",
+            "ADD"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "DiskConfigurationType",
+            "modelAsString": true
+          }
+        },
+        "storageWorkloadType": {
+          "description": "Storage workload type.",
+          "enum": [
+            "GENERAL",
+            "OLTP",
+            "DW"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "StorageWorkloadType",
+            "modelAsString": true
+          },
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ]
+        },
+        "enableStorageConfigBlade": {
+          "description": "Enable SQL IaaS Agent storage configuration blade in Azure Portal.",
+          "default": "false",
+          "type": "boolean",
+          "x-ms-mutability": [
```

Comment:
all values that are not a secret should be readable.  Why wouldn't this value be in PUT/GET responses?